File: Benchmarks/ConcatenatorComparison.cs
using System;
using System.Text.Patterns;
using System.Text.RegularExpressions;
using BenchmarkDotNet.Attributes;

namespace Benchmarks {
	[ClrJob, CoreJob, CoreRtJob]
	[MemoryDiagnoser]
	public class ConcatenatorComparison {

		readonly Pattern Concatenator = (Pattern)"Hello" & ' ' & "World";

		readonly Regex Regex = new Regex("^Hello World");

		[Params("Hello World", "Hello", "Failure")]
		public String Source { get; set; }

		[Benchmark(Baseline = true)]
		public Result Stringier() => Concatenator.Consume(Source);

		[Benchmark]
		public Match MSRegex() => Regex.Match(Source);

	}
}

File: Benchmarks.FSharp/ConcatenatorComparison.fs
namespace Benchmarks

open System.Text.Patterns
open FParsec
open BenchmarkDotNet.Attributes
open BenchmarkDotNet.Diagnosers

[<ClrJob>]
[<MemoryDiagnoser>]
type ConcatenatorComparison() =
    let pattern = "Hello" >> ' ' >> "World"
    let parsec = attempt (pstringCI "Hello" .>>. pchar ' ' .>>. pstringCI "World")

    [<Params("Hello World", "Hello")>]
    member val Source = "" with get, set

    [<Benchmark>]
    member this.PatternConsume() = pattern.Consume(this.Source)

    [<Benchmark(Baseline = true)>]
    member this.ParsecRun() = run parsec this.Source

